### ISSUE TITLE
Prodigy infotech ml task 1

### ISSUE THREAD

**Issue by Anushalatha** [Anushalatha/Prodigy-infotech---Machine-learning--task-1#1]
import numpy as np

class LinearRegression:
    def fit(self, X, y):
        # Add intercept term to X
        X = np.concatenate((np.ones((X.shape[0], 1)), X), axis=1)
        
        # Calculate weights using the normal equation: w = (X^T * X)^-1 * X^T * y
        self.weights = np.linalg.inv(X.T @ X) @ X.T @ y

    def predict(self, X):
        # Add intercept term to X
        X = np.concatenate((np.ones((X.shape[0], 1)), X), axis=1)
        
        return X @ self.weights

# Example usage:
if __name__ == "__main__":
    # Example dataset: square footage and number of bedrooms
    X = np.array([[1000, 2],
                  [1500, 3],
                  [2000, 3],
                  [2500, 4],
                  [3000, 4]])
    y = np.array([300000, 400000, 500000, 600000, 700000])

    # Instantiate and fit the model
    model = LinearRegression()
    model.fit(X, y)

    # Predict prices for new data
    new_data = np.array([[1800, 3],
                         [2200, 4]])
    predicted_prices = model.predict(new_data)
    print("Predicted prices:", predicted_prices)
